ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceWithReferenceExprFixBase.fs

```diff
@@ -0,0 +1,32 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+
+[<AbstractClass>]
+type ReplaceWithReferenceExprFixBase(expr: IFSharpExpression, refExprName, refExprFullName) =
```

Comment:
Are there going to be other inheritors?

I think it would be easier to just use extracted logic (like `deletePatternsFromEnd`) directly from `ExecutePsiTransaction` in each of the two quick fixes.